index.bs

```diff
@@ -630,7 +589,199 @@ self.onmessage = async function(e) {
+    buffer pool allocated to the process is exhausted, standard tee() may be used.
+
+# Security and Privacy considerations # {#security-considerations}
+
+This API defines a {{MediaStreamTrack}} source and a {{MediaStreamTrack}} sink.
+The security and privacy of the source ({{VideoTrackSource}}) relies
+on the same-origin policy. That is, the data {{VideoTrackSource}} can
+make available in the form of a {{MediaStreamTrack}} must be visible to
+the document before a {{VideoFrame}} or {{AudioData}} object can be constructed
+and pushed into the {{MediaStreamTrackGenerator}}. Any attempt to create
+{{VideoFrame}} or {{AudioData}} objects using cross-origin data will fail.
+Therefore, {{MediaStreamTrackGenerator}} does not introduce any new
+fingerprinting surface.
+
+The {{MediaStreamTrack}} sink introduced by this API ({{MediaStreamTrackProcessor}})
+exposes {{MediaStreamTrack}} the same data that is exposed by other
+{{MediaStreamTrack}} sinks such as WebRTC peer connections, Web Audio
+{{MediaStreamAudioSourceNode}} and media elements. The security and privacy
+of {{MediaStreamTrackProcessor}} relies on the security and privacy of the
+{{MediaStreamTrack}} sources of the tracks to which {{MediaStreamTrackProcessor}}
+is connected. For example, camera, microphone and screen-capture tracks
+rely on explicit use authorization via permission dialogs (see
+[[MEDIACAPTURE-STREAMS]] and [[MEDIACAPTURE-SCREEN-SHARE]]),
+while element capture and {{VideoTrackSource}}
+rely on the same-origin policy.
+
+A potential issue with {{MediaStreamTrackProcessor}} is resource exhaustion.
+For example, a site might hold on to too many open {{VideoFrame}} objects
+and deplete a system-wide pool of GPU-memory-backed frames. UAs can
+mitigate this risk by limiting the number of pool-backed frames a site can
+hold. This can be achieved by reducing the maximum number of buffered frames
+and by refusing to deliver more frames to {{MediaStreamTrackProcessor/readable}}
+once the budget limit is reached. Accidental exhaustion is also mitigated by
+automatic closing of {{VideoFrame}} and {{AudioData}} objects once they
+are written to a {{VideoTrackSource}}.
+
+# Backwards compatibility with earlier proposals # {#backwards-compatibility]}
+
+This section is informative.
+
+Previous proposals for this interface had an API like this:
+
+<pre class="idl">
+[Exposed=Window,DedicatedWorker]
+interface MediaStreamTrackGenerator : MediaStreamTrack {
+    constructor(MediaStreamTrackGeneratorInit init);
+    attribute WritableStream writable;  // VideoFrame or AudioFrame
+};
+
+dictionary MediaStreamTrackGeneratorInit {
+  required DOMString kind;
+};
+</pre>
+This interface had the generator for the MediaStreamTrack being an instance of
+a MediaStreamTrack rather than containing one.
+
+This interface can be shimmed on top of VideoTrackSource like this:
+
+<pre>
+
+Javascript goes here when written
+
+</pre>
```

Comment:
I don't agree, for entirely practical reasons: Leaving this shim in here will allow pepole to use the new API shape with existing browsers, letting them early-adopt the new API without waiting for its implementation.

Reply:
Hm. This can be left to an advice note when shipping the new interface in preference to the older one.
I can leave it out; the reason to include it would be to prove by code that the new interface can do everything the old interface could.